ionoscloud/resource_dbaas_mariadb_cluster.go

```diff
@@ -0,0 +1,238 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/dbaas/mariadb"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils/constant"
+)
+
+func resourceDBaaSMariaDBCluster() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: mariaDBClusterCreate,
+		DeleteContext: mariaDBClusterDelete,
+		ReadContext:   mariaDBClusterRead,
+		UpdateContext: mariaDBClusterUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: mariaDBClusterImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"mariadb_version": {
+				Type:             schema.TypeString,
+				Description:      "The MariaDB version of your cluster.",
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringInSlice([]string{"10.6"}, true), validation.StringIsNotWhiteSpace)),
```

Comment:
this validation would mean that if/when the version changes or a new one is added, we need to do a new release of tf to support it.